ocaml/xapi-idl/network/network_interface.ml

```diff
@@ -81,6 +81,31 @@ module Unix = struct
       }
 end
 
+module Macaddr = struct
+  include Macaddr
+
+  let typ_of =
+    Rpc.Types.Abstract
+      {
+        aname= "macaddr"
+      ; test_data= [Macaddr.of_string_exn "ca:fe:ba:be:ee:ee"]
+      ; rpc_of= (fun t -> Rpc.String (Macaddr.to_string t))
+      ; of_rpc=
+          (function
+          | Rpc.String s ->
+              Ok (Macaddr.of_string_exn s)
```

Comment:
to_octets has a more compact notation, I'd rather use that. The of_ function can raise, transform it to an Error instead:
```suggestion
      ; rpc_of= (fun t -> Rpc.String (Macaddr.to_octets t))
      ; of_rpc=
          (function
          | Rpc.String s -> (
            try Ok (Macaddr.of_octets_exn s)
            with e ->
              Error
                (`Msg
                  (Printf.sprintf "typ_of_macaddr: %s" (Printexc.to_string e))
                  )
          )
```